docs/adrs/0002-uds-core-functional-layers.md

```diff
@@ -0,0 +1,73 @@
+
+Today, we publish both the official UDS Core package and a UDS Slim Dev package, in addition to dev and demo UDS Bundles. The UDS Slim Dev package is a stripped-down version of UDS Core that includes limited services meant for testing UDS Packages. We have also learned that some teams outside of UDS need similar functionality, but slightly different parts of UDS Core, such as keycloak + authsvc.
+
+## Decision
+
+We will replace the existing 2-package (one dev and one prod) structure with a layered set of packages within the existing monorepo. This will allow users to choose the components they need for their specific use case, while maintaining a single source of truth for all UDS Core components. We will not break these out by applications as Big Bang did, but instead by layers of functionality for related/integrated components. This allows more flexibility for which functionality is provided by UDS Core and which is provided by other external packages.
+
+The functional layers will be:
+
+1. UDS Core Base (Istio + Pepr w/UDS Operator & UDS Policies)
+2. UDS Core Identity and Authorization (Keycloak + AuthService)
+3. UDS Core Metrics (Metrics Server)
+4. UDS Core Monitoring (Prometheus + Grafana)
+5. UDS Core Logging (Promtail + Loki)
+6. UDS Core Runtime Security (NeuVector)
+7. UDS Core Backup and Restore (Velero)
+8. UDS Core (combination of all layers)
+
+These layers can be combined as needed, with UDS Core Base serving as the foundation for all deployments. The UDS Core package will continue to offer the complete suite of tools. The only change in the UDS Core repo will be the addition of new package definitions for each functional layer. Versioning will be managed at the monorepo level, with each layer being versioned in lockstep with the rest of UDS Core.
+
+We will still publish the UDS Slim Dev bundle for testing UDS Packages. This will be a combination of UDS Core Base, UDS Core Identity and Authorization, and UDS-K3d. We will not publish other bundles beyond UDS Slim Dev and UDS Core Demo. Only the UDS Core zarf package will publish combined layers.
+
+## Consequences
+
+### Positive
+
+- Increased flexibility for users to choose components based on their needs
+- Simplified versioning and dependency management by keeping everything in one repo
+- Easier coordination of changes across layers
+- Maintains a single source of truth for all UDS Core components
+- Enables both testing environments and production edge deployments
+
+### Negative
+
+- Need for more comprehensive documentation on layer combinations and usage
+- Testing needs to be expanded to cover common layer combinations
+- Need to address dependencies and configurations shared across layers
```

Comment:
Maybe doesn't need to be in this ADR since it's a proposal for CLI, but I think it's worth considering if uds-cli should handle dependencies/ordering at the bundle layer rather than core needing to think about that aspect. For example if a bundle creator made a bundle like this:

```yaml
packages:
  - core-identity
  - core-monitoring
  - core-base
```

CLI should be smart enough to either error out on the ordering (core-base should come first) or reorder the bundle so that it would work. Any thoughts on this @UncleGedd? Very connected to other discussions on the idea of coupling CLI closer with Core.

Reply:
Yeah just mentioned in another comment--we need a way at the bundle level to verify these very simple dependencies, I'd be fine if that was hard-coded in UDS CLI for the time being.